File: leetcode/algorithms/longest-palindromic-substring.go
package main

func isPalindrome(s string) bool {
	for i := 0; i < len(s)/2; i++ {
		if s[i] != s[len(s)-1-i] {
			return false
		}
	}
	return true
}

func longestPalindrome(s string) string {
	max_len := 0
	max_str := ""
	for i := 0; i < len(s); i++ {
		for j := i + 1; j <= len(s); j++ {
			if isPalindrome(s[i:j]) {
				if j-i > max_len {
					max_len = j - i
					max_str = s[i:j]
				}
			}
		}
	}
	return max_str
}

func main() {
	println(longestPalindrome("babad"))
	println(longestPalindrome("cbbd"))
	println(longestPalindrome("a"))
	println(longestPalindrome("ac"))
}

File: leetcode/algorithms/longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        max_len = 0
        max_str = ""
        for i in range(len(s)):
            for j in range(i + 1, len(s) + 1):
                if s[i:j] == s[i:j][::-1]:
                    if len(s[i:j]) > max_len:
                        max_len = len(s[i:j])
                        max_str = s[i:j]
        return max_str


s = Solution()
print(s.longestPalindrome('cbabadcbbabbcbabaabccbabc'))
